packages/api-client-core/src/support.ts

```diff
@@ -206,19 +234,15 @@ export const getNonUniqueDataError = (modelApiIdentifier: string, fieldName: str
     `More than one record found for ${modelApiIdentifier}.${fieldName} = ${fieldValue}. Please confirm your unique validation is not reporting an error.`
   );
 
-export const getNonNullableError = (response: Result & { fetching: boolean }, dataPath: string[], throwOnEmptyData = false) => {
+export const getNonNullableError = (response: Result & { fetching: boolean }, dataPath: string[]) => {
   if (response.fetching) {
     return;
   }
   const result = get(response.data, dataPath);
-  const edges = get(result, ["edges"]);
-  const dataArray = edges ?? result;
-  if (result === undefined) {
+  if (result === undefined || result === null) {
     return new GadgetInternalError(
       `Internal Error: Gadget API didn't return expected data. Nothing found in response at ${dataPath.join(".")}`
     );
-  } else if (result === null || (throwOnEmptyData && Array.isArray(dataArray) && dataArray.length === 0)) {
-    return new GadgetInternalError(`Internal Error: Gadget API returned no data at ${dataPath.join(".")}`);
   }
 };
```

Comment:
Shouldn't this be the same logic as `assertOperationSuccess`:
1. If the `data` path is `undefined`, throw an internal error because this should never happen
2. If the `data` path is `null` throw a not found error
3. If the `data` path is an empty array and `throwOnEmptyData` is true, throw a not found error

Reply:
I removed the `throwOnEmptyData` param because it actually never gets called in the code base.

Also at first glance `undefined` and `null` smelled falsy so I thought it'd be more correct if they throw the same InternalError (The API isn't quite clear what findOne, findMany, findFirst can potentially return) but now that I think of it GraphQL returns null if nothing is returned for a field. So I believe your thinking is right and the user should be returned the error that the record was simply not found.

Reply:
cc: @angelini lmk your thoughts as well. What @scott-rc mentioned seems to be the most appropriate way of handling this - as calls to this function (e.g. through errorIfDataAbsent https://github.com/gadget-inc/js-clients/blob/d28e8ecdef5db7b85cb235ed0fff0798ebe23ccb/packages/react/src/useFindOne.ts#L74) can expect to pass a null result set to check against which is not an InternalError but an error on their part that they should probably know about.